numba/npyufunc/_internal.c

```diff
@@ -104,6 +105,557 @@ PyTypeObject PyUFuncCleaner_Type = {
+    PyCFunction ufunc_at;
+} ufunc_dispatch = {NULL, NULL, NULL, NULL, NULL};
+
+static int
+init_ufunc_dispatch(void)
+{
+    int result = 0;
+    PyMethodDef * crnt = PyUFunc_Type.tp_methods;
+    const char * crnt_name = NULL;
+    for (; crnt->ml_name != NULL; crnt++) {
+        crnt_name = crnt->ml_name;
+        switch(crnt_name[0]) {
+        case 'a':
+            if (strncmp(crnt_name, "at", 3) == 0) {
+                ufunc_dispatch.ufunc_at = crnt->ml_meth;
+            } else if (strncmp(crnt_name, "accumulate", 11) == 0) {
+                ufunc_dispatch.ufunc_accumulate =
+                    (PyCFunctionWithKeywords)crnt->ml_meth;
+            } else {
+                result = -1;
+            }
+            break;
+        case 'o':
+            if (strncmp(crnt_name, "outer", 6) == 0) {
+                ufunc_dispatch.ufunc_outer =
+                    (PyCFunctionWithKeywords)crnt->ml_meth;
+            } else {
+                result = -1;
+            }
+            break;
+        case 'r':
+            if (strncmp(crnt_name, "reduce", 7) == 0) {
+                ufunc_dispatch.ufunc_reduce =
+                    (PyCFunctionWithKeywords)crnt->ml_meth;
+            } else if (strncmp(crnt_name, "reduceat", 9) == 0) {
+                ufunc_dispatch.ufunc_reduceat =
+                    (PyCFunctionWithKeywords)crnt->ml_meth;
+            } else {
+                result = -1;
+            }
+            break;
+        default:
+            result = -1; /* Unknown method */
+        }
+        if (result < 0) break;
+    }
+    if (result == 0) {
+        /* Sanity check. */
+        result = ((ufunc_dispatch.ufunc_reduce != NULL) &&
+                  (ufunc_dispatch.ufunc_accumulate != NULL) &&
+                  (ufunc_dispatch.ufunc_reduceat != NULL) &&
+                  (ufunc_dispatch.ufunc_outer != NULL) &&
+                  (ufunc_dispatch.ufunc_at != NULL));
+    }
+    return result;
+}
+
+static PyObject *
+dufunc_reduce(PyDUFuncObject * self, PyObject * args, PyObject *kws)
+{
+    return ufunc_dispatch.ufunc_reduce((PyObject*)self->ufunc, args, kws);
+}
+
+static PyObject *
+dufunc_accumulate(PyDUFuncObject * self, PyObject * args, PyObject *kws)
+{
+    return ufunc_dispatch.ufunc_accumulate((PyObject*)self->ufunc, args, kws);
+}
+
+static PyObject *
+dufunc_reduceat(PyDUFuncObject * self, PyObject * args, PyObject *kws)
+{
+    return ufunc_dispatch.ufunc_reduceat((PyObject*)self->ufunc, args, kws);
+}
+
+static PyObject *
+dufunc_outer(PyDUFuncObject * self, PyObject * args, PyObject *kws)
+{
+    return ufunc_dispatch.ufunc_outer((PyObject*)self->ufunc, args, kws);
+}
+
+static PyObject *
+dufunc_at(PyDUFuncObject * self, PyObject * args)
+{
+    return ufunc_dispatch.ufunc_at((PyObject*)self->ufunc, args);
+}
+
+static PyObject *
+dufunc__compile_for_args(PyDUFuncObject * self, PyObject * args,
+                        PyObject * kws)
+{
+    PyErr_SetString(PyExc_NotImplementedError,
```

Comment:
I beg to differ.  I want to know when my code has somehow called an abstract method.